components/eam/src/physics/crm/samxx/CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ file(GLOB SAMXX_SRC ${CXX_SRC} ${F90_SRC})
 
 # samxx will be a static library
 add_library(samxx STATIC ${SAMXX_SRC})
-if ("${ARCH}" STREQUAL "CUDA")
+if ("${YAKL_ARCH}" STREQUAL "CUDA")
```

Comment:
I'm not sure why you need the YAKL_ARCH var, and not just do 
```
if (USE_CUDA)
```

Reply:
I'll change this to the simpler logic as you suggested. In other projects, I'm used to using YAKL_ARCH because I don't have a USE_CUDA variable defined like we do in E3SM.

Reply:
Makes sense. As I said, not a big deal. Though less cmake vars to  track is always better. :)

Reply:
Should it be `if (USE_CUDA)` or if `(${USE_CUDA})` ? Or does it matter?

Reply:
If you use recent enough cmake (I think >3.8 maybe? which should be the case everywhere by now...) the former is preferable. The latter would cause a warning, telling you to do `if (VAR_NAME)` rather than `if( ${VAR_NAME} )`.